sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -173,6 +186,12 @@ export class BatchingReceiver extends MessageReceiver {
         }
       });
 
+      cleanupAbortSignal = checkAndRegisterWithAbortSignal(
+        () => finalAction(),
```

Comment:
Why are we calling `finalAction()` when the abort signal is fired instead of throwing the AbortError?

Reply:
I was keeping with the sentiment that we've got in the connectionErrorHandler and returning the messages we've got so far, rather than just error out. In receiveAndDelete mode this is particularly important. Less so in peekLock.

TBH, it's inconsistent but it seems like it's favorable in this spot.

@bterlson  - curious what your arch take on this is. Would it be better to be consistent and throw an AbortError here or would it be okay to treat abort as a soft cancel of the current operation when you have partial results?